third-party/py/setuptools/BUCK

```diff
@@ -1,17 +1,2 @@
-python_library(
-  name = 'pkg_resources',
-  base_module = '',
-  srcs = [
-    'pkg_resources.py',
-  ],
-  visibility = [
-    'PUBLIC',
-  ],
-)
-
-export_file(
-  name = 'pkg_resources.py',
-  visibility = [
-    'PUBLIC',
-  ],
-)
+# Note: you shouldn't depend on pkg_resources directly,
+# as it's included in every pex by make_pex.
```

Comment:
So this is where it gets a bit...hand-wavy. There are two explanations for each mode in which pex.py runs:
 - repo mode - In this case, pex.py changes sys.path to ensure it imports pkg_resources in third-party at runtime, so it doesn't need the dependency explicitly.
 - pex mode - in this case pex.py does NOT use the pkg_resources that exist in the repo (i.e. it does not use the dependency) - it uses whatever it was compiled with. That is, building buck.pex (say A) from a buck.pex (say B) will include pkg_resources(B) in buck.pex(A). This was already the case before, so it's not new behavior, but the intent was to clarify it.